### PR TITLE
Traceback logfile expand path

### PR DESCRIPTION
**Pull request by aweltsch**
I moved `built_ins.expand_path` to `tools.expand_path`. I also changed the implementation of
`tools.expandpath` (now `tools._expandpath`) to use `tools.expand_path`. To preserve the distinct behaviour, I added the argument `expand_user=True` to `tools.expand_path` which is checked before performing user/tilde expansion.

Still don't know a good way to test this, since we can not guarantee, that some file like `~/xonsh.log` will be writable by the current user.

### CODE REVIEW

File: xonsh/tools.py [hunk 1]
```diff
@@ -81,14 +81,33 @@ def __init__(self, returncode, command, output=None, stderr=None,
         self.completed_command = completed_command
 
 
-def expandpath(path):
+def expand_path(s, expand_user=True):
```
Comment (mitnk) [inline xonsh/tools.py]: Ideas on adding test cases(for expand_user part): like `p = expand_path('~/foo/bar.txt'); assert p.endswith('/foo/bar.txt') and not p.startswith('~')`.

**Comment by mitnk** [comment]
Thanks @aweltsch, I think this PR is good to go if you could add some basic unit tests for `tools.expand_path()`.

**Comment by scopatz** [comment]
ping @aweltsch , any movement on @mitnk's tests request? It would be nice to merge this in so we have it for the next micro release.

**Comment by aweltsch** [comment]
@scopatz , sorry I was occupied with a lot of stuff for university. I will continue to work on it today.

**Comment by aweltsch** [comment]
I added some tests that test different combinations of EXPAND_ENV_VARS and expand_user. I hope those test cases are enough for now. I'm not sure why circleci fails. I can't see a relation to my changes.

**Comment by scopatz** [comment]
Hi @aweltsch - that is a bug that has been fixed on master and is not your fault. Could you merge from master please? That should clear this up.

**Comment by aweltsch** [comment]
Merge is done.

**Comment by mitnk** [comment]
@aweltsch Thanks a lot for spending time on fixing this issue!

**Pull request merged**